[mlir][spirv] Fix tablegen generator script's stripping of prefixes

andfau-amd opened the pull request:
This script looks for existing definitions with the `SPIRV_` prefix, so that it can preserve them when updating the file. When the commit 2d628330482e49d36744cb8f3fb5047cfeae6c56 changed the prefix from `SPV_`, the number of characters to strip from matched names was not updated, which broke this feature. This commit fixes remaining cases that weren't fixed by 339c87a8a086347bd8b5aae8b5bc43fc1c155cc1.

The relationship of this script to the files it is meant to maintain is still bitrotten in other ways.

kuhar reviewed (commented):
Could we base these on some string length computation instead of using concrete constants?

andfau-amd commented:
> Could we base these some string length computation instead of using concrete constants?

Sure. That was actually done by the previous commit that fixed just a single instance, I'll copy its approach.

kuhar reviewed (approved):
Nice, thanks for fixing this!

kuhar commented:
@andfau-amd seems like we need to run the formatter on this

andfau-amd committed:
[mlir][spirv] Fix tablegen generator script's stripping of prefixes

This script looks for existing definitions with the SPIRV_ prefix, so
that it can preserve them when updating the file. When the commit
2d628330482e49d36744cb8f3fb5047cfeae6c56 changed the prefix from SPV_,
the number of characters to strip from matched names was not updated,
which broke this feature. This commit fixes remaining cases that weren't
fixed by 339c87a8a086347bd8b5aae8b5bc43fc1c155cc1.

The relationship of this script to the files it is meant to maintain is
still bitrotten in other ways.

andfau-amd commented:
Looks like everything is good now.
